module/zfs/dsl_scan.c

```diff
@@ -3356,6 +3357,27 @@ dsl_scan_sync(dsl_pool_t *dp, dmu_tx_t *tx)
 	if (spa->spa_syncing_txg < spa->spa_first_txg + SCAN_IMPORT_WAIT_TXGS)
 		return;
 
+	/*
+	 * zfs_scan_suspend_progress can be set to disable scan progress.
+	 * We don't want to spin the txg_sync thread, so we add a delay
+	 * here to simulate the time spent doing a scan. This is mostly
+	 * useful for testing and debugging.
+	 */
+	if (zfs_scan_suspend_progress) {
+		uint64_t scan_time_ns = gethrtime() - scn->scn_sync_start_time;
+		int mintime = (scn->scn_phys.scn_func == POOL_SCAN_RESILVER) ?
+		    zfs_resilver_min_time_ms : zfs_scrub_min_time_ms;
+
+		while (zfs_scan_suspend_progress &&
+		    !txg_sync_waiting(scn->scn_dp) &&
+		    !spa_shutting_down(scn->scn_dp->dp_spa) &&
+		    NSEC2MSEC(scan_time_ns) < mintime) {
+			delay(hz);
```

Comment:
Since we're already sleeping for a full second here on every iteration, what do you think about using `zfs_sleep_until` instead and simply sleeping for the full `mintime`.

Reply:
didn't know that existed. I'll make the change

Reply:
Actually, now that I'm looking at it I don't like that. I don't think it matters a lot, but that would cause every sync task / txg_wait_synced() to wait up to `mintime` ns. That seems unnecessary and like it could have some negative consequences for code paths that perform sync tasks while this flag is active.

Reply:
OK, that's fine since this code is just for testing.  Though I'll note that for the scrub case since `zfs_scrub_min_time_ms = 1000` it's going to functionally be the same unless we use say `MSEC_TO_TICK(10)` for the sleep delay.